### PR TITLE
Clean-up unified pipelines.

### PR DESCRIPTION
**Pull request by mitchdenny**
This PR removes the client.yml pipeline that was in use before unified pipelines was rolled out. As part of this clean-up we will also be doing the following:

- [x] Removing the internal ```net - client``` pipeline.
- [x] Removing the public ```net - client - ci``` pipeline.
- [x] Remove the net client release pipeline.
- [x] What else do we need to do here?

### CODE REVIEW

**Comment by weshaggard** [comment]
@chidozieononiwu do you know if there are any parts that still require the full client build?

**Comment by chidozieononiwu** [comment]
> @chidozieononiwu do you know if there are any parts that still require the full client build?

No, I think its good to go.

**Comment by mitchdenny** [comment]
Well if you guys see no issues I'll go ahead and merge this and nuke the pipelines (note I included the old release pipeline as one of the pipelines to get rid of). Or does that need to stay for management plane for now?

**Comment by weshaggard** [comment]
As part of removing these client pipelines I do want to ensure we still have coverage for changes under the eng directory. For .NET I've asked @annelo-msft to add that trigger path for core but for the other languages @mitchdenny can you be sure we have those trigger paths setup?

**Comment by mitchdenny** [comment]
Paths are setup.

File: sdk/core/ci.yml [hunk 1]
```diff
@@ -21,6 +21,7 @@ trigger:
   paths:
     include:
     - sdk/core/
+    - eng/
```
Comment (weshaggard) [inline sdk/core/ci.yml]: this was already done in https://github.com/Azure/azure-sdk-for-net/pull/8670

**Review by weshaggard — APPROVED** [review]
I'm OK with this as long as the .NET folks don't have any need for it.

**Comment by chidozieononiwu** [comment]
I wonder if we need to communicate this to the service teams in some way. We had someone from service bus trying to run these pipelines last week.

**Comment by weshaggard** [comment]
We should also clean-up the badges in the root readme https://github.com/Azure/azure-sdk-for-net/blob/master/README.md which are pointing at this build.

**Comment by weshaggard** [comment]
@mitchdenny another thing I saw earlier is that we have some release pipelines like https://dev.azure.com/azure-sdk/internal/_releaseDefinition?definitionId=63&_a=environments-editor-preview that trigger off of these client builds we need to audit those and see if we move them to the unified pipelines or come up with another option

**Pull request merged**